airbyte-webapp/src/packages/cloud/components/experiments/FreeConnectorProgram/hooks/useFreeConnectorProgram.ts

```diff
@@ -1,21 +1,47 @@
+import { useState } from "react";
+import { useIntl } from "react-intl";
 import { useQuery } from "react-query";
+import { useSearchParams } from "react-router-dom";
+import { useEffectOnce } from "react-use";
+
+import { ToastType } from "components/ui/Toast";
 
 import { useExperiment } from "hooks/services/Experiment";
+import { useNotificationService } from "hooks/services/Notification";
 import { useConfig } from "packages/cloud/services/config";
 import { useDefaultRequestMiddlewares } from "services/useDefaultRequestMiddlewares";
 import { useCurrentWorkspaceId } from "services/workspaces/WorkspacesService";
 
 import { webBackendGetFreeConnectorProgramInfoForWorkspace } from "../lib/api";
 
+export const STRIPE_SUCCESS_QUERY = "fcpEnrollmentSuccess";
+
 export const useFreeConnectorProgram = () => {
   const workspaceId = useCurrentWorkspaceId();
   const { cloudApiUrl } = useConfig();
   const config = { apiUrl: cloudApiUrl };
   const middlewares = useDefaultRequestMiddlewares();
   const requestOptions = { config, middlewares };
   const freeConnectorProgramEnabled = useExperiment("workspace.freeConnectorsProgram.visible", false);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [userDidEnroll, setUserDidEnroll] = useState(false);
+  const { formatMessage } = useIntl();
+  const { registerNotification } = useNotificationService();
 
-  return useQuery(["freeConnectorProgramInfo", workspaceId], () =>
+  useEffectOnce(() => {
+    if (searchParams.has(STRIPE_SUCCESS_QUERY)) {
+      // Remove the stripe parameter from the URL
+      setSearchParams({}, { replace: true });
+      setUserDidEnroll(true);
```

Comment:
`userDidEnroll` is essentially a temporary place for us to know that a user has enrolled, but we haven't done a full re-query of the backend yet so we can show the Toast and remove the enrollment materials, right?

Reply:
"Almost right, but the situation is slightly worse than that. After a user successfully finishes the stripe checkout flow: 
1) Stripe redirects the user directly to the `successUrl` their checkout session was configured with (in this case, the airbyte URL the user came from with `STRIPE_SUCCESS_QUERY` appended)
2) fires off a webhook to update our backend

That means that even if we do a full re-query of the backend right away (which in fact we do, since we're reloading the whole webapp), there's a race condition: we cannot rely on the backend to have already processed the enrollment at the moment we ought to show the user a confirmation.